segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -161,14 +163,32 @@ public void setupAppend(SetupAppend setupAppend) {
                 .whenComplete((attributes, u) -> {
                     try {
                         if (u != null) {
+                            if (u instanceof StreamSegmentSealedException) {
+                                throw u;
+                            }
                             handleException(writer, setupAppend.getRequestId(), newSegment, "setting up append", u);
                         } else {
+                            SegmentProperties sp = this.store.getStreamSegmentInfo(newSegment, TIMEOUT).join();
+                            if (sp.isSealed()) {
+                                throw new StreamSegmentSealedException(newSegment);
+                            }
                             long eventNumber = attributes.getOrDefault(writer, Attributes.NULL_ATTRIBUTE_VALUE);
                             this.writerStates.putIfAbsent(Pair.of(newSegment, writer), new WriterState(eventNumber));
                             connection.send(new AppendSetup(setupAppend.getRequestId(), newSegment, writer, eventNumber));
                         }
                     } catch (Throwable e) {
-                        handleException(writer, setupAppend.getRequestId(), newSegment, "handling setupAppend result", e);
+                        if (e instanceof StreamSegmentSealedException || e.getCause() instanceof StreamSegmentSealedException) {
+                            log.info("Segment '{}' is sealed and {} cannot perform append, disabling caching of attribute value...", newSegment, writer);
```

Comment:
1. This should not be info.
2. "disabling caching of attribute value" is misleading. We are not disabling anything, just passing an arg to a method. Disabling may be perceived as "disabling across the board".

I do not see the value of this log message here since the segment store logs its calls with all args.

Reply:
Yes Andrei